eth/core/chain_reader.go

```diff
@@ -137,8 +142,13 @@ func (bc *blockchain) GetTransaction(
 			txLookupEntry.BlockNum, txLookupEntry.TxIndex, nil
 	}
 
-	// check the block plugin
-	txLookupEntry, err := bc.host.GetBlockPlugin().GetTransactionByHash(txHash)
+	// check if historical plugin is supported by host chain
+	if bc.hp == nil {
+		return nil, common.Hash{}, 0, 0, ErrTxNotFound
```

Comment:
do we want to return a more specific error that is saying that the historical plugin isnt setup?

Reply:
I think saying the historical plugin is oddly specific. From the perspective of getting a tx, you either find it or you don't